frontend/src/app/account/profile/profile.component.ts

```diff
@@ -0,0 +1,241 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, Validators, ReactiveFormsModule, FormControl } from '@angular/forms';
+import { MessageService } from 'primeng/api';
+import { AuthenticationService } from '../../../_services/authentication.service';
+import { User } from '../../../_models/user.model';
+import { CommonModule } from '@angular/common';
+import { ButtonModule } from 'primeng/button';
+import { PasswordModule } from 'primeng/password';
+import { ToastModule } from 'primeng/toast';
+import { DividerModule } from 'primeng/divider';
+import { InputTextModule } from 'primeng/inputtext';
+import { invalidUsernameValidator } from '../_validators/invalid-username.validator';
+import { mismatchPasswordValidator } from '../_validators/mismatch-password.validator';
+import { invalidPasswordValidator } from '../_validators/invalid-password.validator';
+import { lowercasePasswordValidator } from '../_validators/lowercase-password';
+import { uppercasePasswordValidator } from '../_validators/uppercase-password';
+import { numericPasswordValidator } from '../_validators/numeric-password';
+import { specialPasswordValidator } from '../_validators/special-password';
+import { shortPasswordValidator } from '../_validators/short-password';
+import { weakPasswordValidator } from '../_validators/weak-password.validator';
+import { FormUtilsService } from '../../../_services/form.utils.service';
+import { Router } from '@angular/router';
+
+@Component({
+    standalone: true,
+    imports: [
+        CommonModule,
+        ReactiveFormsModule,
+        ButtonModule,
+        PasswordModule,
+        ToastModule,
+        DividerModule,
+        InputTextModule,
+    ],
+    providers: [MessageService],
+    templateUrl: './profile.component.html',
+    styleUrls: ['./profile.component.css'],
+})
+export class ProfileComponent implements OnInit {
+    user: User | null = null;
+    showEditProfile = false;
+    showEditPassword = false;
+    isProcessingEdit = false;
+    isProcessingPassword = false;
+
+    constructor(
+        private authenticationService: AuthenticationService,
+        private messageService: MessageService,
+        private router: Router,
+        public formUtils: FormUtilsService,
+    ) {}
+
+    editProfileForm: FormGroup = new FormGroup({
+        username: new FormControl('', [Validators.required, invalidUsernameValidator()]),
+        email: new FormControl('', [Validators.required, Validators.email]),
+        password: new FormControl(''),
```

Comment:
```suggestion
        password: new FormControl('', [Validators.required]),
```

Reply:
I was thinking user should know their username and attackers will have less information without validators. I guess it the information gained can be seen when attackers create password anyways and this would prevent crazy inputs.

Reply:
Sorry don't quite get this haha